app/Nova/Actions/CreateRemoteAttendanceLink.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+// phpcs:disable Generic.Strings.UnnecessaryStringConcat.Found,Squiz.WhiteSpace.OperatorSpacing.SpacingAfter
+
+namespace App\Nova\Actions;
+
+use App\Models\Attendance;
+use App\Models\RemoteAttendanceLink;
+use App\Nova\RemoteAttendanceLink as NovaRemoteAttendanceLink;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Fields\ActionFields;
+use Laravel\Nova\Fields\Text;
+use Lynndigital\SelectOrCustom\SelectOrCustom;
+
+class CreateRemoteAttendanceLink extends Action
+{
+    /**
+     * Indicates if this action is only available on the resource detail view.
+     *
+     * @var bool
+     */
+    public $onlyOnDetail = true;
+
+    /**
+     * Perform the action on the given models.
+     *
+     * @param \Illuminate\Support\Collection<\App\Models\Team|\App\Models\Event>  $models
+     *
+     * @return array<string,string>
+     */
+    public function handle(ActionFields $fields, Collection $models): array
+    {
+        // Four hours is longer than most meetings, but not too much longer than that.
+        $expiration = Carbon::now()->addHours(4);
+
+        // Only on detail, so it will only ever have one model.
+        $attendable = $models->first();
+
+        $link = new RemoteAttendanceLink();
+
+        $link->attendable_type = get_class($attendable);
+        $link->attendable_id = $attendable->id;
+        $link->secret = hash('sha256', random_bytes(64));
```

Comment:
I like this approach better ⬇️
https://github.com/RoboJackets/apiary/blob/239405760dd64900156af7c3f79f9c29dd173400/app/Http/Controllers/SquareCheckoutController.php#L213
(I know this was copied from the previous implementation but it's a good opportunity to swap it.)